auto_nag/scripts/inactive_ni_pending.py

```diff
@@ -0,0 +1,184 @@
+                    requestee_bugs[flag["requestee"]] = [bugid]
+                else:
+                    requestee_bugs[flag["requestee"]].append(bugid)
+
+        user_activity = UserActivity()
+        inactive_users = user_activity.check_users(requestee_bugs.keys())
+        selected_bugs = []
+        for requestee, bugids in requestee_bugs.items():
+            if requestee in inactive_users:
+                selected_bugs.extend(bugids)
+
+        for bugid in set(bugs.keys()) - set(selected_bugs):
+            del bugs[bugid]
+
+        for bug in bugs.values():
+            bug["inactive_ni"] = [
+                {
+                    "id": flag["id"],
+                    "requestee": flag["requestee"],
+                    "requestee_status": user_activity.get_string_status(
+                        inactive_users[flag["requestee"]]
+                    ),
+                }
+                for flag in bug["needinfo_flags"]
+                if flag["requestee"] in inactive_users
+            ]
+            bug["inactive_ni_count"] = len(bug["inactive_ni"])
+            self.add_action(bug)
+
+    def is_recent_bug(self, bug):
+        return lmdutils.get_date_ymd(bug["creation_time"]) >= self.__recent_bug_limit
+
+    def add_action(self, bug):
+        users_num = len(set([flag["requestee"] for flag in bug["inactive_ni"]]))
+
+        if (
+            (
+                bug["priority"] in HIGH_PRIORITY
+                or bug["severity"] in HIGH_SEVERITY
+                or self.is_recent_bug(bug)
+            )
+            and not utils.is_no_assignee(bug["triage_owner"])
+            and not any(
+                [
+                    flag["requestee"] == bug["triage_owner"]
+                    for flag in bug["inactive_ni"]
+                ]
+            )
+        ):
+            autofix = {
+                "flags": [
+                    {
+                        "id": flag["id"],
+                        "requestee": bug["triage_owner"],
+                    }
+                    for flag in bug["inactive_ni"]
+                ],
+                "comment": {
+                    "body": f"Redirect needinfo pending on inactive {utils.plural('user', users_num)} to the triage owner.",
+                },
+            }
+        else:
+            autofix = {
+                "flags": [
+                    {
+                        "id": flag["id"],
+                        "status": "X",
+                    }
+                    for flag in bug["inactive_ni"]
+                ],
+                "comment": {
+                    "body": f"Clear needinfo pending on inactive {utils.plural('user', users_num)}.",
+                },
+            }
+
+        self.add_prioritized_action(bug, bug["triage_owner"], autofix=autofix)
+
+    def get_bug_sort_key(self, bug):
+        return (
+            bug["priority"] not in HIGH_PRIORITY,
+            bug["severity"] not in HIGH_SEVERITY,
```

Comment:
Aren't priority and severity part of get_sort_by_bug_importance_key?
Couldn't this be then:
```
return (
    utils.get_sort_by_bug_importance_key(bug),
    not self.is_recent_bug(bug)
)
```

Reply:
Yes, it is part of `get_sort_by_bug_importance_key()`, however, we use boolean values here to avoid having bugs with P3 priority before bugs with S1 severity.  I improved the logic in b543087714597a65959c1b4d02ec01800c392fa7 to match the needinfo forwarding condition.